pytorch_lightning/trainer/trainer.py

```diff
@@ -498,13 +499,39 @@ def fit(
 
         if self._state != TrainerState.INTERRUPTED:
             self._state = TrainerState.FINISHED
+
+        self._set_running_stage(None)
+
         return results or 1
 
+    def _set_running_stage(self, stage):
+        model_ref = self.get_model()
+
+        if stage is None:
+            self._running_stage = stage
+            model_ref.running_stage = stage
+            return
+
+        # todo: clean up this routing mess.
+        if self._running_stage == RunningStage.TESTING:
+            stage = RunningStage.TESTING
+
+        # WARNING: With predicting,
+        # trainer _running_state should be RunningStage.TESTING
+        # however, the model running_stage should be RunningStage.PREDICTING or None
+        if model_ref is not None:
+            if self._predicting:
+                model_ref.running_stage = RunningStage.PREDICTING
+            else:
+                model_ref.running_stage = stage
```

Comment:
```suggestion
            model_ref.running_stage = self._running_stage
```

Reply:
This won't work. It is a hack to make everything work. I will clean this in another PR.